msteams-platform/apps-in-teams-meetings/build-tabs-for-meeting.md

```diff
@@ -176,26 +176,133 @@ TeamsJS is a rich library used to create Tabs using JavaScript. Use the latest T
 
+## App caching
+
+App caching helps you to improve subsequent launch time of the apps that are loaded in the meeting side panel.
+
+> [!NOTE]
+>
+> * Currently, app caching is available only in [public developer preview](~/resources/dev-preview/developer-preview-intro.md).
+> * App caching is supported only for tabs loaded in the meeting side panel in Teams desktop client.
+
+### Enable app caching
+
+To enable app caching in your meeting side panel, follow the steps:
+
+1. Call `teamsCore.registerBeforeUnloadHandler` and `teamsCore.registerOnLoadHandler` APIs.
+
+1. Use `contentUrl` and `entityId` passed into the load handler to route to the correct page within your app and invoke `notifySuccess/notifyFailure` to notify Teams client that the app initialization flow is complete.
+
+   * [contentUrl](../tabs/how-to/create-tab-pages/configuration-page.md#modify-or-remove-a-tab): Add content page URL.
+   * [entityId](../tabs/how-to/create-tab-pages/configuration-page.md#modify-or-remove-a-tab): Add a unique identifier.
+
+1. Dispose resources and perform any cleanup needed in the `beforeUnload` handler, then invoke the `readyToUnload` callback to notify Teams client that the app unload flow is complete.
+
+The following is the flow diagram of the first launch of an app that wants to opt into app caching (register the `load` or `beforeUnload` on the first launch of the app):
+
+:::image type="content" source="../assets/images/saas-offer/first-launch-app.png" alt-text="This screenshot shows the flow of the first launch of the app in meeting side panel.":::
+
+The following is the flow diagram of the launch of cached app:
+
+:::image type="content" source="../assets/images/saas-offer/cached-launch-app.png" alt-text="This screenshot shows the flow of the cached launch of the app in meeting side panel.":::
+
+When you opt into app caching, the webview that is used to host the embedded app is reused as users navigate to different instances of the app within a window. The webview used to host the app is hidden when the users leave the app and shown when the users return to the app. When the app is cached, any audio that is playing is muted.
+
+> [!NOTE]
+> If the app caching isn't enabled, the webview is recreated every time the users leave and return to the app.
+
+There are multiple reasons for an app to not get cached or get removed from the cache, some of the reasons are (numbers here are subject to change):
+
+* If the system memory load is high, the app is removed from the cache.
+* If the app exceeds maximum cache size, the app is removed from the cache.
+* If the user doesn't return to the app within 20 minutes, the app is removed from the cache.
+* The app isn't cached if Teams doesn't receive the `readyToUnload` signal from the app within 30 seconds after sending the `beforeUnload` notification.
+* App caching is disabled if the system memory is less than 4 GB.
```

Comment:
Pls change to: : (yeah want to add back the available memory bit)
App caching is disabled if the system memory is less than 4 GB or if available memory is less than 1GB on WIndows/512 MB on Mac

Reply:
Updated